### PR TITLE
Pull Request: Bot Updates

### PR DESCRIPTION
**Pull request by ghost**
- [x] Untested
- [ ] Tested

**Pull Request Information**
> This pull request brings some configuration updates, formatting, package fixes, and more...

**Pull Request User**
> User: NFGamerMC

### CODE REVIEW

**Comment by Thunder33345** [comment]
i doubt the gitignore is really needed we can add more as we encounter
and i dont think it will work with removed const
i doubt it would work without bot.run
as for --harmony, what's the reasoning behind it
for dependencies, wouldnt it auto inherit devDependencies?

i think of using the "do what you want" public license

awaiting for @HimbeersaftLP to join so i can actually request review

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,67 @@
-#Gitnore file
+
+# Created by https://www.gitignore.io/api/node
```
Comment (Thunder33345) [inline .gitignore:2]: Might just be best not to brute force ignore everything when there's no thing

File: bot.js [hunk 2]
```diff
@@ -1,137 +1,120 @@
-const Discord = require('discord.js');
-const bot = new Discord.Client();
-const util = require('util')
-const config = require('./config.json')
-
-var readyspam = 0;
-var firstrun = 1;
-var emode = false;
-var token = config.token;
-
```
Comment (Thunder33345) [inline bot.js:10]: i think these vars are necessary for it to run

File: bot.js [hunk 3]
```diff
@@ -1,137 +1,120 @@
...
-  console.log('Everything connected!');
-  if(readyspam == 0){
-    readyspam = 1;
-    setTimeout(function(){
-      readyspam = 0;
-    }, 3000);
-  }else{
-    console.error("Stopping due to client-ready spam, please restart the bot!");
-    process.exit(1);
-  }
```
Comment (Thunder33345) [inline bot.js:21]: might be best just keep this

File: bot.js [hunk 4]
```diff
@@ -1,137 +1,120 @@
...
-    .catch(function(r){
-      console.error("Login not successful! " + r);
-    });
-  firstrun = 0;
-}else{
-  console.log("Not logging in again for preventing bot token reset!");
+bot.on( 'message', msg => {
+	if ( msg.author.id !== bot.user.id ) return;
+	if ( msg.content.startsWith( prefix + 'eval' ) ) { // TODO: better command handling
+		var code = msg.content.substr( 7 );
```
Comment (Thunder33345) [inline bot.js:4]: 7 might not work depending on prefix length

File: config_example.json [hunk 5]
```diff
@@ -1,7 +1,5 @@
 {
   "__comment":"Example config, please copy and paste this as config.json",
-  "prefix": "..",
-  "__comment":"TODO feature",
-
-  "token": "Your Discord bot accounts token",
+  "prefix": "Enter the prefix you wish to use, such as --",
```
Comment (Thunder33345) [inline config_example.json:3]: just keep this as ".."

File: config_example.json [hunk 6]
```diff
@@ -1,7 +1,5 @@
...
+  "token": "Enter your User Token, you can find that by searching 'Discord User Token'"
```
Comment (Thunder33345) [inline config_example.json:4]: just use "false"
they should know how to get their own token already if they wish to use this

File: package.json [hunk 7]
```diff
@@ -1,19 +1,30 @@
...
+    "start": "node --harmony bot.js"
   },
-  "author": "Thunder33345, forked off HimbeersaftLP",
-  "license": "dunno ¯\\_(ツ)_/¯",
+  "author": "Thunder33345",
+  "fork": "HimbeersaftLP",
+  "contributors": {
+      "NFGamerMC [hello@nfgamermc.com] (https://nfgamermc.com)"
+  },
+  "license": "ISC",
```
Comment (Thunder33345) [inline package.json:16]: ❌ dont change the license

File: nb.sh [hunk 8]
```diff
@@ -16,7 +16,7 @@ start(){
 	tmux send-keys -t $TMS:0 "cd $DIR" C-m
 	tmux send-keys -t $TMS:0 "echo Waiting for 3 seconds..." C-m
 	tmux send-keys -t $TMS:0 "sleep 3" C-m
-	tmux send-keys -t $TMS:0 "node bot.js" C-m
+	tmux send-keys -t $TMS:0 "node --harmony bot.js" C-m
```
Comment (Thunder33345) [inline nb.sh:19]: why is --harmony needed?
why current setup wont work?

File: package.json [hunk 9]
```diff
@@ -1,19 +1,30 @@
...
-    "start": "node bot.js"
+    "nm": "nodemon --harmony bot.js",
+    "test": "node --harmony bot.js",
+    "start": "node --harmony bot.js"
   },
-  "author": "Thunder33345, forked off HimbeersaftLP",
-  "license": "dunno ¯\\_(ツ)_/¯",
+  "author": "Thunder33345",
+  "fork": "HimbeersaftLP",
+  "contributors": {
```
Comment (Thunder33345) [inline package.json:13]: i doubt this will be feasible in the future
i dont fancy having an endless list of contributors

File: package.json [hunk 10]
```diff
@@ -1,19 +1,30 @@
...
+      "NFGamerMC [hello@nfgamermc.com] (https://nfgamermc.com)"
+  },
+  "license": "ISC",
   "dependencies": {
     "discord.js": "^11.1.0",
     "erlpack": "github:hammerandchisel/erlpack",
     "node-opus": "^0.2.6",
+    "uws": "^0.14.5"
+  },
+    "devDependencies": {
```
Comment (Thunder33345) [inline package.json:23]: any reason for this and "nodemon"?

**Review by Thunder33345 — CHANGES_REQUESTED** [review]

**Review by HimbeersaftLP — CHANGES_REQUESTED** [review]
Ewww no, recommended js formatting is *two spaces* dude

**Comment by Thunder33345** [comment]
Yes spaces
but like no more @HimbeersaftLP "real" issues?

**Comment by ghost** [comment]
I'll test and fix when home

File: package.json (hunk 7 above)
Comment (ghost) [reply package.json:16]: That is a blank licence genius

File: nb.sh (hunk 8 above)
Comment (HimbeersaftLP) [reply nb.sh:19]: For now its unneeded I think, but maybe we will need it in the future, who knows?

File: bot.js (hunk 2 above)
Comment (HimbeersaftLP) [reply bot.js:10]: Yes they are indeed

**Pull request closed without merging**